dom/src/MainDOMSource.ts

```diff
@@ -56,157 +57,206 @@ const eventTypesThatDontBubble = [
-  let result = false;
-  if (typeof options.useCapture === `boolean`) {
-    result = options.useCapture;
-  }
-  if (eventTypesThatDontBubble.indexOf(eventType) !== -1) {
-    result = true;
-  }
-  return result;
-}
+const STRING_TYPE = `string`;
+
+export class MainDOMSource extends DOMSource implements Sourceable {
+  private _namespace: Array<string>;
+  private _rootElement$: Stream<Element>;
+  private _sanitation$: Stream<{}>;
+  private _isolateModule: IsolateModule;
+  private _eventDelegators: Map<string, EventDelegator>;
+
+  constructor(options: MainDOMSourceOptions) {
+    super(options);
+    this._namespace = options.namespace || [];
+    this._rootElement$ = options.rootElement$;
+    this._sanitation$ = options.sanitation$;
+    this._isolateModule = options.isolateModule;
+    this._eventDelegators = options.eventDelegators;
 
-export class MainDOMSource implements DOMSource {
-  constructor(private _rootElement$: Stream<Element>,
-              private _sanitation$: Stream<{}>,
-              private _runStreamAdapter: StreamAdapter,
-              private _namespace: Array<string> = [],
-              public _isolateModule: IsolateModule,
-              public _delegators: Map<string, EventDelegator>,
-              private _name: string) {
     this.__JANI_EVAKALLIO_WE_WILL_MISS_YOU_PLEASE_COME_BACK_EVENTUALLY = true;
   }
 
-  elements(): any {
-    let output$: Stream<Element | Array<Element>>;
-    if (this._namespace.length === 0) {
-      output$ = this._rootElement$;
-    } else {
-      const elementFinder = new ElementFinder(
-        this._namespace, this._isolateModule
-      );
-      output$ = this._rootElement$.map(el => elementFinder.call(el));
-    }
-    const runSA = this._runStreamAdapter;
-    const out: DevToolEnabledSource = runSA.remember(
-      runSA.adapt(output$, xsSA.streamSubscribe)
+  elements(): DevToolEnabledSource {
+    const output$: Stream<Element | Array<Element>> = this.outputStream();
+    const runStreamAdapter = this._runStreamAdapter;
+    const out: DevToolEnabledSource = runStreamAdapter.remember(
+      runStreamAdapter.adapt(output$, xsAdapter.streamSubscribe)
     );
-    out._isCycleSource = this._name;
+    out._isCycleSource = this._driverKey;
+
     return out;
   }
 
-  get namespace(): Array<string> {
-    return this._namespace;
+  private outputStream(): Stream<Element | Array<Element>> {
+    if (this._namespace.length === 0) { return this._rootElement$; }
+    const elementFinder = new ElementFinder(this._namespace, this._isolateModule);
+
+    return this._rootElement$.map(element => elementFinder.find(element));
   }
 
   select(selector: string): DOMSource {
-    if (typeof selector !== 'string') {
-      throw new Error(`DOM driver's select() expects the argument to be a ` +
+    if (typeof selector !== STRING_TYPE) {
+      throw new Error(`DOM driver’s select() expects the argument to be a ` +
```

Comment:
It’s an apostrophe; not a quote. An apostrophe is only used *within* or at the very end of a word - it is part of the word. Single quotes are only used *around* words - they come in pairs, and are not part of any word.